### ISSUE TITLE
Status bar is overpopulated with information already on the screen

### ISSUE THREAD

**Issue by SylvainCorlay** [jupyterlab/jupyterlab#11547]
It seems that a lot of the information currently shown in the status bar is not really necessary since it is already elsewhere on the page.

<img width="898" alt="Screenshot 2021-11-25 at 11 22 17" src="https://user-images.githubusercontent.com/2397974/143424127-faeb7986-827e-47b2-ad5c-bd01f24b8628.png">

We could easily remove:
 - kernel status,
 - notebook name,
 - the button that opens the left panel for open tabs, kernels, and terminal.

Then we could start using that real estate for other information.

**Comment by krassowski** [comment]
As a general rule of thumb, I would advocate making it user-configurable rather than removing things altogether. Also, some things are useful in multi-panel mode but not in Simple mode - those could be shown/hidden depending on the mode.

Quick comments:
-  +1 for shortening or removing the kernel name (maybe kernelspaecs could provide a short version of the name?)
- kernel status as what is after pipe `|` like `Disconnected`, `Inactive`, etc is not shown anywhere else; personally I would prefer to:
   - keep it, or
   - show it somewhere else prominently, and/or
   - only show it when different from `Idle` or `Busy`
- `Mode:` prefix could be removed/replaced by an icon
- notebook name is very useful when you have multiple notebooks open. I would highly prefer to keep it when in multi-panel mode. +1 for hiding it when in `Simple` mode
- sorry about the LSP verbosity; we are open to suggestions for a shorter text but I could not find a good one.
- "the button that opens the left panel for open tabs, kernels, and terminal." - what does it do in the Simple mode? If it exits simple mode or does nothing, it should be hidden in the Simple mode.

**Comment by SylvainCorlay** [comment]
> "the button that opens the left panel for open tabs, kernels, and terminal." - what does it do in the Simple mode? If it exits simple mode or does nothing, it should be hidden in the Simple mode.

It is there in simple mode, which still has the side panel, so it behaves exactly the same as in the normal mode.

I think that we should remove this button altogether. Another issue with it is that it does not have a toggling behavior (you can't use it hide the panel).

> kernel status as what is after pipe | like Disconnected, Inactive, etc is not shown anywhere else; p

It is there also less evident in the kernel status icon in the upper right corner of notebooks if you hover on it.

> notebook name is very useful when you have multiple notebooks open. I would highly prefer to keep it when in multi-panel mode. +1 for hiding it when in Simple mode

Do you think that anyone reads that name to determine which tab is open in the dock area?

**Comment by krassowski** [comment]
> Do you think that anyone reads that name to determine which tab is open in the dock area?

Yes, I do (sometimes), because finding this out from other places can be difficult:

1) the styles for currently active tab (I mean the blue bar over the tab) is only very subtly different from the currently visible tab (but maybe it's only this way for users with limited colour perception):

   ![Screenshot from 2021-11-25 16-46-39](https://user-images.githubusercontent.com/5832902/143478605-e7661575-e6b6-48cb-84bc-423eb81cf153.png)

2) In my typical layout I usually only see first three-four letters (or even less during deadlines when multitasking more) on the tab name; there is not much other indication on which of the tabs is active and I really don't like when I start typing or shifting focus in a wrong one (a visceral thing learned after multiple issues cause by mis-perception of which tab is open). This is especially relevant when notebooks have the same prefix (e.g. just after duplicating a notebook, or when using step prefix notation like: `1_prepare_data.ipynb`, `2_do_some_analysis.ipynb`). But maybe I am not a typical user - maybe others have more tidy workspaces ;).

    ![Screenshot from 2021-11-25 16-53-43](https://user-images.githubusercontent.com/5832902/143479332-278616c0-77c0-4079-8219-91436a8e5cbf.png)

**Comment by krassowski** [comment]
> It is there also less evident in the kernel status icon in the upper right corner of notebooks if you hover on it.

I don't think this is very user-friendly, especially for new users (I know it is available on hover but this is not something most new users know). I believe it was more user-friendly in classic notebook:

![Screenshot from 2021-11-25 16-50-51](https://user-images.githubusercontent.com/5832902/143480367-241561c9-50eb-40f7-96fb-b958dd997542.png)

**Comment by SylvainCorlay** [comment]
This one button can probably be removed. It opens the side panel for running kernels and terminals, and it cannot close it.

<img width="86" alt="Screenshot 2021-12-08 at 16 52 00" src="https://user-images.githubusercontent.com/2397974/145239420-3b17d3eb-83f0-46a1-9e64-ac294faa2edd.png">

**Comment by krassowski** [comment]
> This one button can probably be removed. It opens the side panel for running kernels and terminals, and it cannot close it.

Looking at the thumbs up I it seems there is a consensus and hiding it (by default/when there is not enough space) would be fine.

However, this button could be useful because the default behaviour is to keep kernel running when users close the tab, so one can be running out of memory in a long-lived session just because they _thought_ the kernels get closed. This indicator may be teaching users about this quirk.

If we hide/remove these icons I would like to consider adding add a tour (https://github.com/jupyterlab/jupyterlab/issues/9812) which would activate on first time user closes a notebook (or is just a part of the initial tour) teaching users about this behaviour. At a minimum, along with the PR hiding/removing the icons there should come an issue to track/discuss how to compensate for it.

If we really want to remove (rather than hide), maybe we could move it to [jupyter-resource-usage](https://github.com/jupyter-server/jupyter-resource-usage) (and make it configurable) or to a new extension? Not a strong feeling here, just a suggestion.

**Comment by krassowski** [comment]
Regarding removal of the name of the notebook: I opened #11716 - once addressed there will be one argument less for keeping the notebook name on the status bar :)

**Comment by Shruteek** [comment]
Has there been any update on being able to modify items on the status bar? I am currently running JupyterLab 4.0.11 and have found that the information on the status bar is so dense that I am usually not able to see most of it due to truncation. I have tried to find options & modify my user config files but have not found any way to remove items from the status bar.

**Comment by krassowski** [comment]
> I am currently running JupyterLab 4.0.11

The has been some adjacent work, JupyterLab 4.1 now reduces the amount of information shown in very narrow windows https://github.com/jupyterlab/jupyterlab/pull/14854.

> the information on the status bar is so dense that I am usually not able to see most of it due to truncation

Can you post a screenshot? I am curious how much of this comes from extensions (I know that jupyterlab-lsp is one culprit) and how much is just core jupyterlab.

**Comment by Shruteek** [comment]
I apologize for not responding - GitHub did not notify me! The issue disappeared after a lot of trial and error, but I do not remember exactly how - I did at one point update to the current JupyterLab (I am now using version 4.2.4), so that may have to do with it. I actually did not have jupyterlab-lsp enabled, I don't believe, though it was installed. I appreciate the response; if anyone else has a similar issue, I'm sorry I can't give a clear solution!

**Comment by jtpio** [comment]
FYI I just opened https://github.com/jupyterlab/jupyterlab/pull/16846 to hide the running sessions status bar item by default, but still allow for showing via a user setting. And added it to the `4.4.0` milestone so it can be considered for the next minor release.

**Comment by krassowski** [comment]
What about auto-hiding the terminals part when there are no running terminals? This would reduce the terminals + kernels item by half for many users.
